### PR TITLE
[ticket/13815] Event parameters in posting have no effect

### PR DESCRIPTION
**Pull request by RMcGirr83**
The submit and error parameters within the core.posting_modify_submit_post_before/after have no effect as the event is only allowed if submit and not error

[PHPBB3-13815] (https://tracker.phpbb.com/browse/PHPBB3-13815)

### CODE REVIEW

**Comment by RMcGirr83** [comment]
!set Event

**Comment by VSEphpbb** [comment]
The purpose of these events is to define errors before and after performing the post action, so `error` is definitely needed to be modified by an extension.

`submit` also may not seem needed, but that doesn't mean an extension might find it useful to have that var available either.

**Comment by RMcGirr83** [comment]
https://tracker.phpbb.com/browse/PHPBB3-13815?focusedCommentId=56254&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-56254

File: phpBB/posting.php [hunk 1]
```diff
@@ -1371,10 +1371,8 @@
 			* @var	string	post_author_name	Author name for guest posts
 			* @var	bool	update_message		Boolean if the post message was changed
 			* @var	bool	update_subject		Boolean if the post subject was changed
-			* @var	bool	submit		Whether or not the form has been submitted
-			* @var	array	error		Any error strings; a non-empty array aborts form submission.
-			*				NOTE: Should be actual language strings, NOT language keys.
 			* @since 3.1.0-RC5
+			* @updated 3.1.5-RC1
```
Comment (nickvergessen) [inline phpBB/posting.php]: its changed instead of updated

Comment (nickvergessen) [inline phpBB/posting.php]: please also explain what happened.

File: phpBB/posting.php [hunk 2]
```diff
@@ -1388,8 +1386,6 @@
 				'post_author_name',
 				'update_message',
 				'update_subject',
-				'submit',
-				'error',
```
Comment (nickvergessen) [inline phpBB/posting.php]: in case of before, I'd prefer to fix error.
So an extension can set it and the submit is aborted.

Comment (RMcGirr83) [inline phpBB/posting.php]: @nickvergessen not understanding what you are saying.

File: phpBB/posting.php [hunk 3]
```diff
@@ -1413,10 +1409,8 @@
 			* @var	bool	update_message		Boolean if the post message was changed
 			* @var	bool	update_subject		Boolean if the post subject was changed
 			* @var	string	redirect_url		URL the user is going to be redirected to
-			* @var	bool	submit		Whether or not the form has been submitted
-			* @var	array	error		Any error strings; a non-empty array aborts form submission.
-			*				NOTE: Should be actual language strings, NOT language keys.
 			* @since 3.1.0-RC5
+			* @updated 3.1.5-RC1
```
Comment (nickvergessen) [inline phpBB/posting.php]: its changed instead of updated
please also explain what happened.

**Comment by javiexin** [comment]
!unset WIP

**Comment by javiexin** [comment]
What is pending here?

**Comment by RMcGirr83** [comment]
Not sure myself.

**Comment by RMcGirr83** [comment]
!unset WIP

**Comment by RMcGirr83** [comment]
Hrmph

File: phpBB/posting.php [hunk 4]
```diff
@@ -1371,10 +1371,8 @@
 			* @var	string	post_author_name	Author name for guest posts
 			* @var	bool	update_message		Boolean if the post message was changed
 			* @var	bool	update_subject		Boolean if the post subject was changed
-			* @var	bool	submit		Whether or not the form has been submitted
-			* @var	array	error		Any error strings; a non-empty array aborts form submission.
-			*				NOTE: Should be actual language strings, NOT language keys.
 			* @since 3.1.0-RC5
+			* @change 3.1.5-RC1 remove submit and error from event  Submit and Error are checked previously prior to running event
```
Comment (marc1706) [inline phpBB/posting.php]: Should be 3.1.6-RC1

File: phpBB/posting.php (hunk 2 above)
Comment (marc1706) [inline phpBB/posting.php]: @nickvergessen The error variable only has the mentioned effect prior to the if statement this event is in. In here, this doesn't have any effect.

File: phpBB/posting.php [hunk 5]
```diff
@@ -1413,10 +1409,8 @@
 			* @var	bool	update_message		Boolean if the post message was changed
 			* @var	bool	update_subject		Boolean if the post subject was changed
 			* @var	string	redirect_url		URL the user is going to be redirected to
-			* @var	bool	submit		Whether or not the form has been submitted
-			* @var	array	error		Any error strings; a non-empty array aborts form submission.
-			*				NOTE: Should be actual language strings, NOT language keys.
 			* @since 3.1.0-RC5
+			* @change 3.1.5-RC1 remove submit and error from event  Submit and Error are checked previously prior to running event
```
Comment (marc1706) [inline phpBB/posting.php]: 3.1.6-RC1

**Comment by RMcGirr83** [comment]
Have no idea why the build is failing for such a simple commit.

**Comment by marc1706** [comment]
This commit shouldn't have the (reverted ...) in the commit message: https://github.com/RMcGirr83/phpbb/commit/3102eda44ee0ea1b5f4d0c4c9ede0c731bdc67d0

**Comment by RMcGirr83** [comment]
@marc1706 I'll just do a new pull request.  This one is a mess.

**Comment by RMcGirr83** [comment]
https://github.com/phpbb/phpbb/pull/3780

**Pull request closed without merging**